### DISCUSSION TITLE
How to increase the size of SLabelView

### DISCUSSION THREAD

**Discussion by we-wake-act21** [eclipse-glsp/glsp#798] [General]
I want to set the size for the SLabelView to be equal to the Node (RectangularNodeView). Is there a way to implement this?

**Comment by planger** [comment]
On a first glance this doesn't seem easy to be easy to do, but maybe I'm missing something.

I haven't tried it but it could work to do the following in a customized `SLabelView`:
* Determine the width of the container
* Determine the width of the label
* Compute the `<factor>` of the container width relative to the label width
* Add a `scale(<factor>)` to the text element `transform` attribute

Note however that this could get problematic if Sprotty layouters are invovled, as it messes with the sizes during the first hidden rendering pass.

Another approach may be to use a [custom layouter](https://www.eclipse.org/glsp/documentation/clientlayouting/#custom-layouter) that scales the label according to the container size.

Yet another approach could be to use a [postprocessor](https://github.com/eclipse/sprotty/blob/master/packages/sprotty/src/base/views/vnode-postprocessor.ts).

Sorry that I can't give a more concrete suggestion. I didn't have this requirement yet and would have to try a few things myself.

**Reply by we-wake-act21** [reply]
<h3>So in context of what i am trying to do here is </h3>

![Screenshot 2022-11-21 152216](https://user-images.githubusercontent.com/116638890/203020612-237e7baa-5482-4e1e-afab-7c0e84ce7716.png)

<h3>As shown in the image, the text label is only contained in the highlighted space, so when I am clicking on the remaining area of Rectangular Node (Shown below) I am unable to change the label text.</h3>

![Screenshot 2022-11-21 152243](https://user-images.githubusercontent.com/116638890/203021654-d01cb039-de55-4778-ab2f-4ddc8c2db1fe.png)

is it possible to combine the label and node together.?
